scripts/tools/mod_tools.js

```diff
@@ -348,6 +350,38 @@ function hashModName(modName) {
     return hash.substring(0, 16);
 }
 
+// Returns a gulp pattern array, used for matching file types when copying files over from '.temp' folder
+function buildBundleGulp(buildDir, extConfigName){
+    let gulp_array = [
+        buildDir + '/*([0-f])',
+        '!' + buildDir + '/dlc'
+    ]
+    
+    if (extConfigName != 'bundleExtension'){
+        gulp_array[0] += config.get(extConfigName);
+    }
+
+    return gulp_array
+}
+
+// Returns the bundle's file stream and initiates the process of copying the bundle file to the buildDir
+function bundleStreamer(extConfigName, buildDir, useNewFormat, modName, reject) {
+    let bundleStream = vinyl.src(
+        buildBundleGulp(buildDir, extConfigName),
+        { base: buildDir })
+        .pipe(rename(p => {
+
+            if (!useNewFormat) {
+                p.basename = hashModName(modName);
+            }
+
+            p.extname = config.get(extConfigName);
```

Comment:
And then here you can check if it doesn't have an extension, then add it, otherwise keep the existing one.

Reply:
Thanks for the suggestion! I was focused on discerning what file the type was before piping the contents over and didn't consider doing it inside the pipe.